Gems/Atom/Asset/ImageProcessingAtom/Code/Source/ImageLoader/TIFFLoader.cpp

```diff
@@ -366,25 +368,19 @@ namespace ImageProcessingAtom
                 }
             };
 
-            bool readError = false;
-
             // Loop across the image height, one tile at a time
-            for (uint32_t imageY = 0; (imageY < inputImageHeight) && (!readError); imageY += tileHeight)
+            for (uint32_t imageY = 0; imageY < inputImageHeight; imageY += tileHeight)
             {
                 // Loop across the image width, one tile at a time
                 for (uint32 imageX = 0; imageX < inputImageWidth; imageX += tileWidth)
                 {
                     // Either read in a tile or a scanline
-                    auto readResult = isTiled ?
-                        TIFFReadTile(tif, buf.data(), imageX, imageY, 0, 0) :
+                    auto result = isTiled?
+                        TIFFReadTile(tif, buf.data(), imageX, imageY, 0, 0):
                         TIFFReadScanline(tif, buf.data(), imageY);
 
-                    // Stop processing if there was an error reading the input data.
-                    if (readResult == -1)
-                    {
-                        readError = true;
-                        break;
-                    }
```

Comment:
If we decide to downgrade this to a warning, we should at least clear out buf.data() after reads with errors so that we get deterministic all-black pixels instead of non-deterministic data based on whatever is in the buffer. Or alternatively, if we think libtiff is successfully converting at least some of the data, maybe clear out buf.data() before each read. Ideally, this would also get tested with some corrupt TIFF files to verify that different types of errors don't cause any crashes if processing continues forward.

Also, right now result needs a [[maybe_unused]] on it since it's only being used inside of AZ_Warning.